analyzers/src/SonarAnalyzer.CSharp/Extensions/ISymbolExtensions.cs

```diff
@@ -89,5 +91,28 @@ private static IEnumerable<SyntaxNode> GetDescendantNodes(Location location, Syn
 
             return root.DescendantNodes();
         }
+
+        /// <summary>
+        /// Returns the Identifier token of the first declaring <see cref="SyntaxReference"/>.
+        /// For e.g. an <see cref="IPropertySymbol"/> this is <see cref="PropertyDeclarationSyntax.Identifier"/>.
+        /// </summary>
+        /// <returns>The idenifier token or <see langword="null"/> if there is no DeclaringSyntaxReference or the <see cref="SyntaxNode"/> does not have an Identifier.</returns>
+        public static SyntaxToken? GetFirstIdentifier(this ISymbol symbol) =>
+            symbol.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax().NodeIdentifier();
+
+        /// <summary>
+        /// Returns the Identifier tokens of all declaring <see cref="SyntaxReference"/>s.
+        /// For e.g. an <see cref="INamedTypeSymbol"/> this can be a list of partial <see cref="ClassDeclarationSyntax.Identifier"/>.
+        /// </summary>
+        public static ImmutableArray<SyntaxToken> GetIdentifiers(this ISymbol symbol) =>
```

Comment:
Having this, the other one (`FirstDeclaringReferenceIdentifier`) should be just something like `symbol.DeclaringReferenceIdentifiers().FirstOrDefault()`, shouldn't it?
```suggestion
        public static ImmutableArray<SyntaxToken> DeclaringReferenceIdentifiers(this ISymbol symbol) =>
```